gluten-core/src/main/scala/io/glutenproject/expression/BinaryExpressionTransformer.scala

```diff
@@ -83,6 +90,63 @@ class LikeTransformer(substraitExprName: String,
   }
 }
 
+case class Sha2Transformer(substraitExprName: String,
+                      left: ExpressionTransformer,
+                      right: ExpressionTransformer,
+                      original: Sha2)
+  extends ExpressionTransformer with Logging {
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(GlutenConfig.GLUTEN_CLICKHOUSE_BACKEND)) {
```

Comment:
As proposed by #1439, let's make `doTransform` common to different backends. And move the backend-specific validation logic into the corresponding backend. Thanks!

Reply:
@PHILO-HE I'll finish it ASAP.